src/modules/tools/zprobe/ZGridStrategy.cpp

```diff
@@ -534,8 +543,22 @@ void ZGridStrategy::normalize_grid()
 
 void ZGridStrategy::homexyz()
 {
-    Gcode gc("G28", &(StreamOutput::NullStream));
-    THEKERNEL->call_event(ON_GCODE_RECEIVED, &gc);
+
+  switch(this->home_before_probe) {
+    case NOHOME : return;
+
+    case HOMEXY : {
+                      Gcode gc("G28 X0 Y0", &(StreamOutput::NullStream));
```

Comment:
Ok... I'll fix it again!


  [image: photo]
*Quentin Harley*
Founder, Reprap Morgan 3D
 p:+27120041883 | e:quentin@reprapmorgan.com | w:www.reprapmorgan.com | a:4
Burger Ave, Lyttelton Manor, Centurion

On Thu, Jul 2, 2015 at 9:28 PM Jim Morris <notifications@github.com> wrote:

> In src/modules/tools/zprobe/ZGridStrategy.cpp
> <https://github.com/Smoothieware/Smoothieware/pull/681#discussion_r33813767>
> :
>
> > @@ -534,8 +543,22 @@ void ZGridStrategy::normalize_grid()
> >
> >  void ZGridStrategy::homexyz()
> >  {
> > -    Gcode gc("G28", &(StreamOutput::NullStream));
> > -    THEKERNEL->call_event(ON_GCODE_RECEIVED, &gc);
> > +
> > +  switch(this->home_before_probe) {
> > +    case NOHOME : return;
> > +
> > +    case HOMEXY : {
> > +                      Gcode gc("G28 X0 Y0", &(StreamOutput::NullStream));
>
> no you are wrong it is G28 X0 Y0
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Smoothieware/Smoothieware/pull/681/files#r33813767>.
>